File: javascript-sdk/src/KestraClient.js
import ApiClient from "./ApiClient"

import AppsApi from './api/AppsApi';
import AuditLogsApi from './api/AuditLogsApi';
import AuthsApi from './api/AuthsApi';
import BannersApi from './api/BannersApi';
import BindingsApi from './api/BindingsApi';
import BlueprintTagsApi from './api/BlueprintTagsApi';
import BlueprintsApi from './api/BlueprintsApi';
import ClusterApi from './api/ClusterApi';
import DashboardsApi from './api/DashboardsApi';
import DefaultApi from './api/DefaultApi';
import ExecutionsApi from './api/ExecutionsApi';
import FilesApi from './api/FilesApi';
import FlowsApi from './api/FlowsApi';
import GroupsApi from './api/GroupsApi';
import InvitationsApi from './api/InvitationsApi';
import KVApi from './api/KVApi';
import LogsApi from './api/LogsApi';
import MaintenanceApi from './api/MaintenanceApi';
import MetricsApi from './api/MetricsApi';
import MiscApi from './api/MiscApi';
import NamespacesApi from './api/NamespacesApi';
import PluginsApi from './api/PluginsApi';
import RolesApi from './api/RolesApi';
import SCIMApi from './api/SCIMApi';
import SCIMConfigurationApi from './api/SCIMConfigurationApi';
import SecurityIntegrationsApi from './api/SecurityIntegrationsApi';
import ServicesApi from './api/ServicesApi';
import TenantsApi from './api/TenantsApi';
import TestSuitesApi from './api/TestSuitesApi';
import TriggersApi from './api/TriggersApi';
import UsersApi from './api/UsersApi';
import WorkerGroupsApi from './api/WorkerGroupsApi';


class KestraClient {

    constructor(host, accessToken, username, password) {
        this.apiClient = ApiClient.instance;

        this.apiClient.basePath = host;
        if (accessToken) {
            this.apiClient.authentications.bearerAuth.accessToken = accessToken
        }
        if (username && password) {
            this.apiClient.authentications.basicAuth.username = username;
            this.apiClient.authentications.basicAuth.password = password;
        } else {
            if (username || password) {
                console.error("Username or password is provided, but not both.");
            }
        }

        // Initialize all APIs
        this.appsApi = new AppsApi(this.apiClient);
        this.auditLogsApi = new AuditLogsApi(this.apiClient);
        this.authsApi = new AuthsApi(this.apiClient);
        this.bannersApi = new BannersApi(this.apiClient);
        this.bindingsApi = new BindingsApi(this.apiClient);
        this.blueprintTagsApi = new BlueprintTagsApi(this.apiClient);
        this.blueprintsApi = new BlueprintsApi(this.apiClient);
        this.clusterApi = new ClusterApi(this.apiClient);
        this.dashboardsApi = new DashboardsApi(this.apiClient);
        this.defaultApi = new DefaultApi(this.apiClient);
        this.executionsApi = new ExecutionsApi(this.apiClient);
        this.filesApi = new FilesApi(this.apiClient);
        this.flowsApi = new FlowsApi(this.apiClient);
        this.groupsApi = new GroupsApi(this.apiClient);
        this.invitationsApi = new InvitationsApi(this.apiClient);
        this.kvApi = new KVApi(this.apiClient);
        this.logsApi = new LogsApi(this.apiClient);
        this.maintenanceApi = new MaintenanceApi(this.apiClient);
        this.metricsApi = new MetricsApi(this.apiClient);
        this.miscApi = new MiscApi(this.apiClient);
        this.namespacesApi = new NamespacesApi(this.apiClient);
        this.pluginsApi = new PluginsApi(this.apiClient);
        this.rolesApi = new RolesApi(this.apiClient);
        this.scimConfigurationApi = new SCIMConfigurationApi(this.apiClient);
        this.scimApi = new SCIMApi(this.apiClient);
        this.securityIntegrationsApi = new SecurityIntegrationsApi(this.apiClient);
        this.servicesApi = new ServicesApi(this.apiClient);
        this.tenantsApi = new TenantsApi(this.apiClient);
        this.testSuitesApi = new TestSuitesApi(this.apiClient);
        this.triggersApi = new TriggersApi(this.apiClient);
        this.usersApi = new UsersApi(this.apiClient);
        this.workerGroupsApi = new WorkerGroupsApi(this.apiClient);

    }

}

export default KestraClient;

File: java-sdk/src/main/java/io/kestra/sdk/KestraClient.java
/*
This class is the only class that is not generated.
It is an easy-to-use entry point to create a client and access the different API endpoints.
 */
package io.kestra.sdk;


import io.kestra.sdk.api.*;
import io.kestra.sdk.api.*;
import io.kestra.sdk.internal.ApiClient;

import java.util.Base64;
import java.util.Objects;

/**
 * KestraClient is the main entry point to the Kestra API SDK.
 * You can create a client via its builder, then you can call any of its methods to access API endpoints.
 * <p>
 * For example, to access the Flows endpoint:
 * <pre>
 *     var kestraClient = KestraClient.builder().url("https://my-kestra-instance.org").build();
 *     var flowApi = kestraClient.flows();
 * </pre>
 */
// TODO we may want to generate it via a template
public class KestraClient {
    private final ApiClient apiClient;

    /**
     * Creates a KestraClient builder.
     * Use this build to instantiate a new client.
     */
    public static KestraClientBuilder builder() {
        return new KestraClientBuilder();
    }

    KestraClient(ApiClient apiClient) {
        this.apiClient = apiClient;
    }

    // FIXME we may want to hide some API from the generator (and also from the OpenAPI spec) as some are only there for the UI
    // START -- Individual API

    public AppsApi apps() {
        return new AppsApi(this.apiClient);
    }

    public AuditLogsApi auditLogs() {
        return new AuditLogsApi(this.apiClient);
    }

    public AuthsApi auths() {
        return new AuthsApi(this.apiClient);
    }

    public BannersApi banners() {
        return new BannersApi(this.apiClient);
    }

    public BindingsApi bindings() {
        return new BindingsApi(this.apiClient);
    }

    public BlueprintsApi blueprints() {
        return new BlueprintsApi(this.apiClient);
    }

    public BlueprintTagsApi blueprintTags() {
        return new BlueprintTagsApi(this.apiClient);
    }

    public DashboardsApi dashboards() {
        return new DashboardsApi(this.apiClient);
    }

    // FIXME why is it named default and not invitation?
    public DefaultApi defaultApi() {
        return new DefaultApi(this.apiClient);
    }

    public ExecutionsApi executions() {
        return new ExecutionsApi(this.apiClient);
    }

    public FilesApi files() {
        return new FilesApi(this.apiClient);
    }

    public FlowsApi flows() {
        return new FlowsApi(this.apiClient);
    }

    public GroupsApi groups() {
        return new GroupsApi(this.apiClient);
    }

    public InvitationsApi invitations() {
        return new InvitationsApi(this.apiClient);
    }

    public KvApi kv() {
        return new KvApi(this.apiClient);
    }

    public LogsApi logs() {
        return new LogsApi(this.apiClient);
    }

    public MetricsApi metrics() {
        return new MetricsApi(this.apiClient);
    }

    public MiscApi misc() {
        return new MiscApi(this.apiClient);
    }

    public NamespacesApi namespaces() {
        return new NamespacesApi(this.apiClient);
    }

    public PluginsApi plugins() {
        return new PluginsApi(this.apiClient);
    }

    public RolesApi roles() {
        return new RolesApi(this.apiClient);
    }

    public ScimConfigurationApi scimConfiguration() {
        return new ScimConfigurationApi(this.apiClient);
    }

    public SecurityIntegrationsApi securityIntegrations() {
        return new SecurityIntegrationsApi(this.apiClient);
    }

    public ServicesApi services() {
        return new ServicesApi(this.apiClient);
    }

    public TenantsApi tenants() {
        return new TenantsApi(this.apiClient);
    }

    public TriggersApi triggers() {
        return new TriggersApi(this.apiClient);
    }

    public UsersApi users() {
        return new UsersApi(this.apiClient);
    }

    public WorkerGroupsApi workerGroups() {
        return new WorkerGroupsApi(this.apiClient);
    }

    // END -- Individual API

    /**
     * Builder for a KestraClient.
     */
    public static class KestraClientBuilder {
        private String url = "http://localhost:8080";
        private Auth auth = Auth.BASIC;
        private String token;
        private String username;
        private String password;

        /**
         * The base URL of the Kestra API.
         * Will default to http://localhost:8080 if not set.
         */
        public KestraClientBuilder url(String url) {
            this.url = Objects.requireNonNull(url);
            return this;
        }

        /**
         * Enable Bearer Token authentication.
         */
        public KestraClientBuilder tokenAuth(String token) {
            this.token = Objects.requireNonNull(token);
            this.auth = Auth.TOKEN;

            return this;
        }

        /**
         * Enable Basic authentication.
         */
        public KestraClientBuilder basicAuth(String username, String password) {
            this.username = Objects.requireNonNull(username);
            this.password = Objects.requireNonNull(password);
            this.auth = Auth.BASIC;

            return this;
        }

        public KestraClient build() {
            ApiClient apiClient = new ApiClient();
            apiClient.setBasePath(url);
            switch (auth) {
                case TOKEN: {
                    apiClient.addDefaultHeader("Authorization", "Bearer " + token);
                    break;
                }
                case BASIC: {
                    String basicAuth = username + ":" + password;
                    apiClient.addDefaultHeader("Authorization", "Basic " + new String(Base64.getEncoder().encode(basicAuth.getBytes())));
                    break;
                }
            }

            return new KestraClient(apiClient);
        }
    }

    public enum Auth { NONE, BASIC, TOKEN }
}
